File: modules/core/src/QuantumResult.cpp
#include "miniqbt/interpreter/QuantumResult.hpp"
#include "miniqbt/interpreter/exceptions/InterpreterExceptions.hpp"

using namespace MiniQbt;

QuantumResult::QuantumResult() {
    this->registerName = "ERROR";
}   

QuantumResult::QuantumResult(const std::string& registerName, const std::vector<bool>& data) {
    this->registerName = registerName;
    this->data = data;
}

std::string QuantumResult::getName() const {
    return this->registerName;
}
std::string QuantumResult::dataToString() const {
    std::string res = "";
    for(bool i : this->data) {
        res += (i)? "1" : "0";
    }
    return res;
}

int QuantumResult::registerSize() const {
    return this->data.size();
}

bool QuantumResult::getData(int index) const {
    if(index > this->registerSize()) {
        // todo: make a custom exception
        throw InterpreterOutOfIndexException();
    }
    return this->data[index];
}

File: modules/java/javaBindings/src/QuantumResult.java
package nl.hvanderheijden.miniqbt;

public class QuantumResult {

    private final String registerName;
    private final boolean[] registerData;

    public QuantumResult(final String registerName, boolean[] data ) {
        this.registerName = registerName;
        registerData = new boolean[data.length];
        for(int i = 0; i < data.length; i++) {
            registerData[i] = data[i];
        }
    }

    public String getName() {
        return this.registerName;
    }
    
    @Override
    public String toString() {
        return String.format("%s[%s]", this.registerName, this.dataToString());
    }

    public String dataToString() {
        String result = "";
        for(boolean in : this.registerData) {
            result += (in)? "0" : "1";
        }

        return result;
    }

    public int registerSize() {
        return this.registerData.length;
    }

    public boolean getData(int index) {
        return registerData[index];
    }
}